pydm/widgets/timeplot.py

```diff
@@ -81,24 +80,31 @@ def configure_timer(self):
   
   # Adds a new curve to the current plot
   def addYChannel(self, ychannel='', curve_color=None):
-    # Allocate space for a new data buffer
-    self.num_of_channels = self.num_of_channels + 1
-    new_channel_id = self.num_of_channels
+    if ychannel == '':
+      raise Exception('[Error] Parameter ychannel should be defined for a new curve.')
```

Comment:
What about changing the generic Exception to ValueError?
https://docs.python.org/3/library/exceptions.html#ValueError